ledger/participant-state/kvutils/app/src/main/scala/com/daml/ledger/participant/state/kvutils/app/Config.scala

```diff
@@ -92,249 +92,280 @@ object Config {
       args: collection.Seq[String],
       getEnvVar: String => Option[String] = sys.env.get(_),
   ): Option[Config[Extra]] =
-    parser(name, extraOptions, getEnvVar).parse(args, createDefault(defaultExtra)).flatMap {
-      case config if config.mode == Mode.Run && config.participants.isEmpty =>
-        System.err.println("No --participant provided to run")
-        None
-      case config =>
-        Some(config)
-    }
+    parser(name, extraOptions, getEnvVar)
+      .parse(args, createDefault(defaultExtra))
+      .flatMap {
+        case config if config.mode == Mode.Run && config.participants.isEmpty =>
+          System.err.println("No --participant provided to run")
+          None
+        case config =>
+          Some(config)
+      }
 
   private def parser[Extra](
       name: String,
       extraOptions: OptionParser[Config[Extra]] => Unit,
       getEnvVar: String => Option[String],
   ): OptionParser[Config[Extra]] = {
-    val parser: OptionParser[Config[Extra]] = new OptionParser[Config[Extra]](name) {
-      head(name)
-
-      opt[Map[String, String]]("participant")
-        .unbounded()
-        .text(
-          "The configuration of a participant. Comma-separated pairs in the form key=value, with mandatory keys: [participant-id, port] and optional keys [address, port-file, server-jdbc-url, max-commands-in-flight, management-service-timeout, run-mode, shard-name]"
-        )
-        .action((kv, config) => {
-          val participantId = ParticipantId.assertFromString(kv("participant-id"))
-          val port = Port(kv("port").toInt)
-          val address = kv.get("address")
-          val portFile = kv.get("port-file").map(new File(_).toPath)
-          val runMode: ParticipantRunMode = kv.get("run-mode") match {
-            case None => ParticipantRunMode.Combined
-            case Some("combined") => ParticipantRunMode.Combined
-            case Some("indexer") => ParticipantRunMode.Indexer
-            case Some("ledger-api-server") => ParticipantRunMode.LedgerApiServer
-            case Some(unknownMode) =>
-              throw new RuntimeException(
-                s"$unknownMode is not a valid run mode. Valid modes are: combined, indexer, ledger-api-server. Default mode is combined."
+    val parser: OptionParser[Config[Extra]] =
+      new OptionParser[Config[Extra]](name) {
+        head(name)
+
+        opt[Map[String, String]]("participant")
+          .unbounded()
+          .text(
+            "The configuration of a participant. Comma-separated pairs in the form key=value, with mandatory keys: [participant-id, port] and optional keys [" +
+              "address, " +
+              "port-file, " +
+              "server-jdbc-url, " +
+              "api-server-connection-pool-size" +
+              "indexer-server-connection-pool-size" +
```

Comment:
We really need to make these share a connection pool.

Reply:
I would say it's actually healthy to have them separate. The indexer should be able to push as many ledger state updates to the index db, regardless of how busy the rest of the participant is (either handling command submissions or serving the rest of the ledger api).

Reply:
Then a prioritized connection pool. :upside_down_face: 

I think you're right though; way too much effort and no guarantee of any actual improvement. This is much simpler.